plugins/processors/awsentity/processor.go

```diff
@@ -0,0 +1,413 @@
+
+	rm := md.ResourceMetrics()
+	for i := 0; i < rm.Len(); i++ {
+		var logGroupNames, serviceName, environmentName string
+		var entityServiceNameSource, entityPlatformType string
+		var ec2Info entitystore.EC2Info
+		resourceAttrs := rm.At(i).Resource().Attributes()
+		switch p.config.EntityType {
+		case entityattributes.Resource:
+			if p.config.Platform == config.ModeEC2 {
+				ec2Info := getEC2InfoFromEntityStore()
+				if ec2Info.InstanceID != EMPTY {
+					resourceAttrs.PutStr(entityattributes.AttributeEntityType, entityattributes.AttributeEntityAWSResource)
+					resourceAttrs.PutStr(entityattributes.AttributeEntityResourceType, entityattributes.AttributeEntityEC2InstanceResource)
+					resourceAttrs.PutStr(entityattributes.AttributeEntityIdentifier, ec2Info.InstanceID)
+				}
+				AddAttributeIfNonEmpty(resourceAttrs, entityattributes.AttributeEntityAwsAccountId, ec2Info.AccountID)
+			}
+		case entityattributes.Service:
+			if logGroupNamesAttr, ok := resourceAttrs.Get(attributeAwsLogGroupNames); ok {
+				logGroupNames = logGroupNamesAttr.Str()
+			}
+			if serviceNameAttr, ok := resourceAttrs.Get(attributeServiceName); ok {
+				serviceName = serviceNameAttr.Str()
+			}
+			if environmentNameAttr, ok := resourceAttrs.Get(attributeDeploymentEnvironment); ok {
+				environmentName = environmentNameAttr.Str()
+			}
+			if serviceNameSource, sourceExists := resourceAttrs.Get(entityattributes.AttributeEntityServiceNameSource); sourceExists {
+				entityServiceNameSource = serviceNameSource.Str()
+			}
+
+			entityServiceName := getServiceAttributes(resourceAttrs)
+			entityEnvironmentName := environmentName
+			if (entityServiceName == EMPTY || entityEnvironmentName == EMPTY) && p.config.ScrapeDatapointAttribute {
+				entityServiceName, entityEnvironmentName, entityServiceNameSource = p.scrapeServiceAttribute(rm.At(i).ScopeMetrics())
+				// If the entityServiceNameSource is empty here, that means it was not configured via instrumentation
+				// If entityServiceName is a datapoint attribute, that means the service name is coming from the UserConfiguration source
+				if entityServiceNameSource == entityattributes.AttributeServiceNameSourceUserConfig && entityServiceName != EMPTY {
+					entityServiceNameSource = entityattributes.AttributeServiceNameSourceUserConfig
+				}
+			}
+			if p.config.KubernetesMode != "" {
+				p.k8sscraper.Scrape(rm.At(i).Resource())
+				if p.config.Platform == config.ModeEC2 {
+					ec2Info = getEC2InfoFromEntityStore()
+				}
+
+				if p.config.KubernetesMode == config.ModeEKS {
+					entityPlatformType = entityattributes.AttributeEntityEKSPlatform
+				} else {
+					entityPlatformType = entityattributes.AttributeEntityK8sPlatform
+				}
+
+				podInfo, ok := p.k8sscraper.(*k8sattributescraper.K8sAttributeScraper)
+				// Perform fallback mechanism for service and environment name if they
+				// are empty
+				if entityServiceName == EMPTY && podInfo.Workload != EMPTY {
```

Comment:
nit: We don't check if the type assertion succeeded, so `podInfo` could be nil, but that should never happen.

Reply:
Blocking comment